### ISSUE TITLE
Bug: @typescript-eslint/typedef  - not functioning properly

### ISSUE THREAD

**Issue by n-ochs** [typescript-eslint/typescript-eslint#7212]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.1.6&fileType=.ts&code=MYewdgzgLgBAhjAvDALAbgFAdJWAjJGAcghAFsBTGANzgBsBXCorHaGYQqAJydfHYATQgG8YguFDgAuYqUpQAFgEswAcyIwAvv1wwqyANpE43bkQA0MAIxWAZvQgUAuhiA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6I0gE0QDN0pE0aAHtokcGAC%2BISUA&tsconfig=&tokens=false

### Repro Code

```typescript
const a = 4;

const b = 'some value'

const c = true

const d = { data: 'something' }

const e = ['arr', 1, false]
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/typedef": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

I expected each of the constants declared to be an error due to the eslint rule

### Actual Result

No error appears on any of the constants

**Comment by Josh-Cena** [comment]
By default, this rule checks nothing. You need to use `variableDeclaration: true` you make it work here. Please check the docs: https://typescript-eslint.io/rules/typedef/#options
